src/views/DMInitiative/components/ResourceDrawer.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import Box from "@mui/material/Box";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import { Monster, useDndApiMonsters } from "@services/DndApiService";
+import Drawer from "@components/Drawer";
+import { IconButton, ListItemText, TextField } from "@mui/material";
+import AddIcon from "@mui/icons-material/Add";
+
+type ResourceDrawerProps = {
+  onAdd?: (data: any) => void;
+};
+
+const ResourceDrawer = ({ onAdd }: ResourceDrawerProps) => {
+  const { isLoading, monsters } = useDndApiMonsters();
+
+  const [search, setSearch] = useState<string>();
+
+  if (isLoading || monsters.length === 0) return;
+
+  const handleMonsterAdd = (monster: Monster) => {
+    const monsterItem = {
+      name: monster.name,
+      health: monster.hitPoints,
+      maxHealth: monster.hitPoints,
+      armorClass: monster.armorClass,
+      initiativeBonus: Math.floor((monster.dexterity - 10) / 2),
+    };
+    onAdd?.(monsterItem);
+  };
+
+  return (
+    <Drawer>
```

Comment:
So wierdly enough (even though I'm def seeing it on your vid.) My drawer isn't closing on select or click outside. There should be a onClose() prop for the drawer and isOpen that can hook into <Drawer />

How do you feel about leveraging TS interfaces by using

declare interface ResourceDrawerProps extends DrawerProps 

haven't seen that pattern a ton, but I could just be missing it! was wondering your thoughts on it

Reply:
Also, I can def add that stuff if you're busy, I'm probably going to adding some more to the drawer functionality soon anyways.